search/binary_test.go

```diff
@@ -1,23 +1,45 @@
 package search
 
-import (
-	"testing"
-)
+import "testing"
 
 func TestBinarySearch(t *testing.T) {
 	for _, test := range searchTests {
-		actual := Binary(test.data, test.key, 0, len(test.data)-1)
-		if actual != test.expected {
-			t.Errorf("test %s failed", test.name)
+		actualValue, _ := Binary(test.data, test.key, 0, len(test.data)-1)
```

Comment:
I think you're currently ignoring the errors raised because of the other implementations. could you instead create a new test case locally inside this test function to check the expected errors as well. Going forward it would be good to cover everything implemented. If you can change the other implementations, I'd be happy to review but if not, I suggest that you create a local struct for the test cases and test it on them to make sure everything works as expected 😄

Reply:
I think I will open a new PR to implement error and benchmark test on other search functions, is that ok?